### PR TITLE
bug 1804913: make ipv6 support a little better

### PR DESCRIPTION
**Pull request by deads2k**
Updates the etcd operator to

1. sign for all internal node IP addresses. This will fix issues with nodes having multiple IPs and not being trusted
2. choose the correct preferred ip address based on the networks.config.openshift.io
3. properly escape ipv6 addresses in the static pod and in the internal etcd client

### CODE REVIEW

**Comment by danwinship** [comment]
with git master installer and this branch of cluster-etcd-operator:

```
danw@p50:installer (master)> OPENSHIFT_INSTALL_AZURE_EMULATE_SINGLESTACK_IPV6=true OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE=registry.svc.ci.openshift.org/ipv6/release:4.4.0-0.nightly-2020-02-19-123359-ipv6.4 ./bin/openshift-install --dir /home/danw/clusters/44 create cluster
INFO Credentials loaded from file "/home/danw/.azure/osServicePrincipal.json" 
WARNING Found override for release image. Please be warned, this is not advised 
INFO Consuming Install Config from target directory 
INFO Creating infrastructure resources...         
ERROR                                              
ERROR Warning: "resource_group_name": [DEPRECATED] This field has been deprecated and is no longer used - will be removed in 2.0 of the Azure Provider 
ERROR                                              
ERROR   on ../../../../../../tmp/openshift-install-378083010/main.tf line 166, in resource "azurerm_storage_container" "vhd": 
ERROR  166: resource "azurerm_storage_container" "vhd" { 
ERROR                                              
ERROR (and 3 more similar warnings elsewhere)      
ERROR                                              
ERROR                                              
ERROR Error: Code="OSProvisioningTimedOut" Message="OS Provisioning for VM 'dwinship-ipv6-srpxx-master-0' did not finish in the allotted time. The VM may still finish provisioning successfully. Please check provisioning state later. Also, make sure the image has been properly prepared (generalized).\r\n * Instructions for Windows: https://azure.microsoft.com/documentation/articles/virtual-machines-windows-upload-image/ \r\n * Instructions for Linux: https://azure.microsoft.com/documentation/articles/virtual-machines-linux-capture-image/ " 
ERROR                                              
ERROR   on ../../../../../../tmp/openshift-install-378083010/master/master.tf line 81, in resource "azurerm_virtual_machine" "master": 
ERROR   81: resource "azurerm_virtual_machine" "master" { 
ERROR                                              
ERROR                                              
ERROR                                              
ERROR Error: Code="OSProvisioningTimedOut" Message="OS Provisioning for VM 'dwinship-ipv6-srpxx-master-1' did not finish in the allotted time. The VM may still finish provisioning successfully. Please check provisioning state later. Also, make sure the image has been properly prepared (generalized).\r\n * Instructions for Windows: https://azure.microsoft.com/documentation/articles/virtual-machines-windows-upload-image/ \r\n * Instructions for Linux: https://azure.microsoft.com/documentation/articles/virtual-machines-linux-capture-image/ " 
ERROR                                              
ERROR   on ../../../../../../tmp/openshift-install-378083010/master/master.tf line 81, in resource "azurerm_virtual_machine" "master": 
ERROR   81: resource "azurerm_virtual_machine" "master" { 
ERROR                                              
ERROR                                              
ERROR                                              
ERROR Error: Code="OSProvisioningTimedOut" Message="OS Provisioning for VM 'dwinship-ipv6-srpxx-master-2' did not finish in the allotted time. The VM may still finish provisioning successfully. Please check provisioning state later. Also, make sure the image has been properly prepared (generalized).\r\n * Instructions for Windows: https://azure.microsoft.com/documentation/articles/virtual-machines-windows-upload-image/ \r\n * Instructions for Linux: https://azure.microsoft.com/documentation/articles/virtual-machines-linux-capture-image/ " 
ERROR                                              
ERROR   on ../../../../../../tmp/openshift-install-378083010/master/master.tf line 81, in resource "azurerm_virtual_machine" "master": 
ERROR   81: resource "azurerm_virtual_machine" "master" { 
ERROR                                              
ERROR                                              
ERROR                                              
ERROR Error: "frontend_ip_configuration.1.private_ip_address" is not a valid IPv4 address: "fc00::ffff:ffff:ffff:fffe" 
ERROR                                              
ERROR   on ../../../../../../tmp/openshift-install-378083010/vnet/internal-lb.tf line 6, in resource "azurerm_lb" "internal": 
ERROR    6: resource "azurerm_lb" "internal" {     
ERROR                                              
ERROR                                              
FATAL failed to fetch Cluster: failed to generate asset "Cluster": failed to create cluster: failed to apply using Terraform 
```

networking portion of install-config:
```
clusterNetwork:
  - cidr: fd01::/48
    hostPrefix: 64
machineNetwork:
  - cidr: 10.0.0.0/16
  - cidr: fc00::/48
networkType: OVNKubernetes
serviceNetwork:
  - fd02::/112
```

**Comment by danwinship** [comment]
install failure is probably fallout from https://github.com/openshift/installer/pull/2745. I'm retrying with an older build

**Comment by danwinship** [comment]
> install failure is probably fallout from [openshift/installer#2745](https://github.com/openshift/installer/pull/2745).

(which is 4.5, not 4.4)

File: pkg/dnshelpers/util.go [hunk 1]
```diff
@@ -14,6 +16,89 @@ import (
...
+	for _, currAddress := range node.Status.Addresses {
+		if currAddress.Type == corev1.NodeInternalIP {
+			switch ipFamily {
+			case "tcp4":
+				isIPv4, err := IsIPv4(currAddress.Address)
+				if err != nil {
+					return "", "", err
+				}
+				if isIPv4 {
+					return currAddress.Address, ipFamily, nil
```
Comment (sanchezl) [inline pkg/dnshelpers/util.go]: if an IPv4 address in IPv6 format, `isIPv4` will be true, but you will need the brackets.
`net.IP`'s `String()`  will return a canonical IP address string that will match the ipFamily type.

Comment (danwinship) [reply pkg/dnshelpers/util.go]: node.Status.Addresses should never have IPv4 addresses in IPv6 format (other things will almost definitely break if it does)

**Pull request merged**